File: yukicoder/00005/block.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	var L, N int
	var W [10000]int
	fmt.Scan(&L)
	fmt.Scan(&N)
	for i := 0; i < N; i++ {
		fmt.Scan(&W[i])
	}
	w := W[:N]
	sort.Ints(w)
	sum := 0
	c := 0
	for i := 0; i < N; i++ {
		if sum+W[i] > L {
			break
		}
		sum += W[i]
		c++
	}
	fmt.Println(c)
}

File: yukicoder/00005/block.cpp
#include <algorithm>
#include <iostream>

using namespace std;

int main() {
    int L, N;
    int W[10000];
    cin >> L;
    cin >> N;
    for (int i = 0; i < N; i++) cin >> W[i];
    sort(W, W + N);
    int sum = 0;
    int count = 0;
    for (int i = 0; i < N; i++) {
        if (sum + W[i] > L) break;
        sum += W[i];
        count++;
    }
    cout << count << endl;
    return 0;
}
